### ISSUE TITLE
[form] not working click input after initialise form

### ISSUE THREAD

**Issue by apple-idn** [fomantic/Fomantic-UI#746]
# Bug Report

  [Form] Inside on click form not working

## Steps to reproduce
1.  create input file in field
2. initialize ui.form(if not initialise form working fine)
3. click file not running script
4. before FUI 2.7.5 its working fine
## Expected result
open file it open
## Actual result
not working input click
## Testcase
https://jsfiddle.net/allwe/1bpuv5r8/2/

## Screenshot (when possible)
![]()

## Version
2.7.5

**Comment by lubber-de** [comment]
Regression since adf593733e0ed463dcb18b697f651a5df91143e5 through #661 which implemented the clean/dirty states to a form stopped the event propagation in the w `isDirty` method:

https://github.com/fomantic/Fomantic-UI/blob/7be57fcaaccf7b88f7e85e70ede8c3d5259eed61/src/definitions/behaviors/form.js#L302-L304

If `e.stopImmediatePropagation(); ` is removed, the issue is fixed.

@prudho Do you know why this line was necessary (or how we can fix it otherwise)

**Comment by apple-idn** [comment]
`$('body').on('click', '.dede', function(e) {
  e.preventDefault();
  var himp = $(this).closest('.action');
  himp.find('input[type="file"]').val('');
  himp.find('input[type="file"]').click();
  console.log('welcome dum file')
});
$('body').on('change', 'input[type="file"]', function (e) {
         var ini = $(this);
         console.log(e)
        var himp = ini.closest('.action');
        var nama_file = e.target.files[0].name;
        himp.find('input[name="dum_file"]').val(nama_file);
        });
$('.ui.form').form();`
**but type file on change** not work @lubber-de

**Comment by lubber-de** [comment]
@apple-idn As said, if the above mentioned line is removed, then your issue is fixed.
See your adjusted fiddle https://jsfiddle.net/ea1urjgs/

@prudho Do you remember why the above mentioned stopPropagation was needed in the isDirty method?

**Comment by prudho** [comment]
MMmmmh... I'm not sure, maybe it's to avoid triggering multiple events when more than one item is selected. We should do some tests to be sure enough to remove it.

**Comment by apple-idn** [comment]
thanks @lubber-de **BUT answer not complete** where input type files empty(not execution), where $('body').on('change', 'input[type="file"]', function(e) {..... not running after, same case if only input type file on form i's **EMPTY**
see my fiddle
https://jsfiddle.net/allwe/dftr12Le/4/

**Comment by lubber-de** [comment]
It's fixed now by #767 
@apple-idn 
I adjusted your latest fiddle. The reason, why the file input was not updating the filename was unfortunately because of many bugs in your code
```html
 <input class="dede" type="text" placeholder="Pilih File (*.xlsx)..." readonly="" nama="dum_file">
<!-- name instead of nama -->
```
```javascript
//$($this) has to be $(this)
  var ini = $($this);
  console.log(e);
  var himp = $(this).closest('.action');
  var filename = e.target.files[0].name;
// selector has to be 'input[name="dum_file"]
// filename instead of fileName (typo)
  himp.find('input[type="dum_file"]').val(fileName);
```
I corrected everything here😉 
See https://jsfiddle.net/y0dsjotp/2/

**Comment by apple-idn** [comment]
wow it's very clear and fixed thanks very much @lubber-de
